server/galaxyls/services/tools/macros.py

```diff
@@ -81,6 +81,14 @@ def load_macro_definitions(self, tool_xml: XmlDocument) -> ToolMacroDefinitions:
             macros=macros,
         )
 
+    def get_macro_names(self, tool_xml: XmlDocument) -> Set[str]:
+        tool = GalaxyToolXmlDocument.from_xml_document(tool_xml)
+        macros = self._get_macro_definitions(tool_xml)
+        imported_macro_files = self._get_imported_macro_files_from_tool(tool)
+        for file in imported_macro_files.values():
+            macros.update(file.macros)
```

Comment:
Oo, if there is the same name in multiple macros we should probably issue a warning somewhere, this can be scary.

Reply:
That's a good idea, I'll add it to the [macros support issue](https://github.com/galaxyproject/galaxy-language-server/issues/44) ;) although this will be something for the validation service.